### DISCUSSION TITLE
开启本地搜索后，搜索功能无效

### DISCUSSION THREAD

**Discussion by lvjianzhao** [YunYouJun/hexo-theme-starry#304] [Q&A 疑难杂症]
作者您好，这两天看到您的这个主题，非常喜欢，所以迫不及待的就用上了，但是现在有个问题，就是每次加载页面，控制台会输出如下：
local-search.js:1 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'textContent')
    at local-search.js:1:2024
    at Array.map (<anonymous>)
    at local-search.js:1:1938

同时搜索功能（开启的是本地搜索）无法使用，现象可以到我的博客：https://lvjianzhao.gitee.io/  进一步查看。

我这边排查了：
1、谷歌、火狐、edge三款浏览器（均为最新版）都试过了，同样的现象；
2、也不是一直不能用，有一阵是好的，但后来不知道改过什么导致的还是怎么回事，忽然发现搜索功能不可以使用了；
3、我是个前端小垃圾，实在是能力有限，期盼大佬帮忙看下；
4、报错中提到的local-search.js文件我也从来没改过，本地文件的md5值为：439317010cce7605cef7293f8edf8a60；
5、本地安装的模块列表如下：
``` bash
npm list
hexo-site@0.0.0 D:\blog\lvjianzhao.gitee.io
├── babel-plugin-prismjs@2.1.0
├── hexo-abbrlink@2.2.1
├── hexo-algoliasearch@1.0.0
├── hexo-deployer-git@3.0.0
├── hexo-generator-archive@1.0.0
├── hexo-generator-category@1.0.0
├── hexo-generator-feed@3.0.0
├── hexo-generator-index@2.0.0
├── hexo-generator-search@2.4.3
├── hexo-generator-tag@1.0.0
├── hexo-render-pug@2.1.4
├── hexo-renderer-ejs@2.0.0
├── hexo-renderer-marked@5.0.0
├── hexo-renderer-stylus@2.1.0
├── hexo-server@3.0.0
├── hexo-theme-landscape@0.0.3
├── hexo-theme-yun@1.9.3
├── hexo-wordcount@6.0.1
├── hexo@6.2.0
├── prismjs@1.28.0
└── shelljs@0.8.5
```

期盼回复，万分感谢！！！

**Comment by lvjianzhao** [comment]
我好像知道原因了，因为最初我的本地搜索相关配置如下：
``` bash
search:
  path: search.xml
  field: post
  content: true
```

但实际搜索时，发现由于是全文搜索，导致搜索结果过多（并且巨卡，输入文字后，半天才能显示到搜索框），所以关闭了全文搜索，更改后的配置如下：
``` bash
search:
  path: search.xml
  field: post
  content: false
```


这样就会导致生成的search.xml文件中，每一段entry中没有content这个元素，导致js文件中的代码：`content:e.querySelector("content").textContent`执行时取不到值而报错。希望作者可以在新版本中对这块做个判断或异常处理吧。避免导致整个搜索功能不可用。

**Comment by lvjianzhao** [comment]
对了，之前用hexo-theme-next主题，也是开启的全文搜索，但并没有这种现象：就是键盘打上去要搜索的关键字，要很久之后输入框才能显示关键字（试了下，如果在其他编辑器编辑好要搜索的字，然后粘贴到搜索框中，反应速度还是很快的）。如果作者大佬有精力的话，希望这个问题也查看下。

**Comment by YunYouJun** [comment]
Link #305.
